lib/puppet/provider/service/systemd.rb

```diff
@@ -46,11 +67,20 @@ def status
   end
 
   def enable
+    output = systemctl("unmask", @resource[:name])
```

Comment:
Yeah, I see the dilemma and I'm not sure either.

I like the idea of enable respecting :mask, and I sorta think that part is non-controversial.

But I'm not 100% sure what to do about the default for :mask. On the one hand, clearly the default for :mask should be false for standard uses (e.g. I didn't even know 'masked' was a thing until today). But if someone is using 'masked' to mean hands-off even for puppet ... This of course goes to your first bullet "puppet is authoritative".

Thoughts?

(Btw, I feel like I need to educate myself about usage of 'masked' to get a better feel for this whole topic.)